frontend/src/components/admin/routeEditing.vue

```diff
@@ -0,0 +1,214 @@
+            </div>
+            <div class="field">
+                <label class="label">Schedule Editor</label>
+                <schedule-editor v-model="route.schedule" />
+            </div>
+            <div class="field">
+            <div class="control">
+                <button @click="send" v-if="formValid" :class="{'is-loading': this.sending}" class="button is-info">Save</button>
+            </div>
+            </div>
+
+        </div>
+    </div>
+</div>
+</template>
+<script lang="ts">
+import Vue from 'vue';
+import mapView from '@/components/admin/map.vue';
+import Route from '../../structures/route';
+import scheduleEditor from '@/components/admin/scheduleEditor.vue';
+import AdminServiceProvider from '../../structures/serviceproviders/admin.service';
+import drawRoute from '@/components/admin/drawRoute.vue';
+import * as L from 'leaflet';
+
+// This component is the route editing interface, it creates a new route object on mounted and route change to isolate edits.
+// To use route creation mode pass in a :creation="true" prop
+export default Vue.extend({
+    data() {
+        return {
+            route: new Route(-1, '', '', true, '#ff00ff', 4, [], [], true),
+            routePolyLine: undefined,
+            sending: false,
+            success: false,
+            failure: false,
+        } as {
+            route: Route;
+            routePolyLine: L.Polyline | undefined;
+            sending: boolean;
+            success: boolean;
+            failure: boolean;
+        };
+    },
+    components: {
+        mapView,
+        scheduleEditor,
+        drawRoute,
+    },
+    mounted() {
+        if (this.creation) {
+            return;
+        }
+        // if the routes are not in the state store yet, wait until they are
+        const el = this;
+        if (this.$store.getters.getRoutes.length === 0) {
+            this.$store.subscribe((mutation) => {
+                if (mutation.type === 'setRoutes') {
+                    el.grabMyRoute();
+                }
+            });
+        } else {
+            this.grabMyRoute();
+        }
+    },
+    watch: {
+        $route() {
+            if (this.creation) {
+                return;
+            }
+            this.grabMyRoute();
+        },
+    },
+    computed: {
+        colorValid(): boolean {
+            const validColor = new RegExp('^#(?:[0-9a-fA-F]{3}){1,2}$');
+            return validColor.test(this.route.color);
+        },
+        formValid(): boolean {
+            return this.route.name !== '' && this.colorValid && this.route.width > 0;
+        },
+    },
+    methods: {
+        setPoints(points: Array<{latitude: number, longitude: number}>) {
+            this.route.points = points;
+        },
+        send() {
+            this.sending = true;
+            if (!this.formValid) {
+                return;
+            }
+            if (!this.creation) {
+                AdminServiceProvider.EditRoute(this.route).then(() => {
```

Comment:
Editing/creating routes seems buggy, and I will follow-up with more details.

Reply:
The schedule constraint requires that if `start_day = end_day` then `start_time < end_time`.

Here's [the actual constraint](https://github.com/wtg/shuttletracker/blob/bfa68d4b481b909d7b30c2edfcfdf79f23510076/postgres/route.go#L52):
```
-- Note: active intervals for route schedules for a route cannot wrap around
-- the week boundary. This is for simplicity of implementation in the
-- route_is_active() function.
CHECK (
	(start_day = end_day AND start_time < end_time) OR (start_day < end_day)
)```